### ISSUE TITLE
Running the app in ChromeOS will ignore the query parameter

### ISSUE THREAD

**Issue by kinyat** [GoogleChromeLabs/bubblewrap#422]
**Describe the bug**
https://developers.google.com/web/android/trusted-web-activity/query-parameters

It is suggested by the Google team to add query parameters for tracking TWA. However, we realised that when we run the APP in ChromeOS with the query parameters set in `startUrl` in `twa-manifest.json`, ChromeOS will always ignore the query parameters.

By adding the query parameters also in `start_url` in the webmanifest file. We managed to make the app so that the first run directly after the installation will still ignore the query parameter but the 2nd open and onwards will have the parameter set correctly.

We didn't test it on Android because our APP only supports ChromeOS:

Here is the related part of `twa-manifest.json`

```
{
  "display": "fullscreen",
  "startUrl": "/?utm_source=twa",
  "webManifestUrl": "https://example.com/manifest-twa.webmanifest",
  "isChromeOSOnly": true,
  ...(rests are omitted),
}
```

Here is the related part of our `manifest-twa.webmanifest`

```
{
  "display": "standalone",
  "scope": "./",
  "start_url": "/?utm_source=twa",
  ...(rests are omitted),
}
```

**To Reproduce**
First Scenario:
1. Build the APP with query parameter set in `startUrl` in `twa-manifest.json`
2. Open the APP in ChromeOS via an Android intent (i.e. via the Play Store dialog after installation) 
3. Observe that the query parameter is missing (i.e. window.location.search is missing)
4. When you open the second time via the shelf, observe that query parameter is still missing

Second Scenario:
1. Build the APP with query parameter set in both `startUrl` in `twa-manifest.json` and `start_url` in webmanifest file (e.g. `/?utm_source=twa`)
2. Open the APP in ChromeOS via an Android intent (i.e. via the Play Store dialog after installation) 
3. Observe that the query parameter is missing (i.e. window.location.search is missing)
4. When you open the second time via the shelf, observe that query parameter is set (i.e. window.location.search returns you something)

**Expected behavior**
It should always have the query parameter set when specified in `startUrl` in `twa-manifest.json`

**Desktop:**
 - OS: ChromeOS
 - Browser: Built in browser
- Version: 87

**Comment by andreban** [comment]
@jeevan-shikaram, could it be the case that `start_url` is being extracted from `/res/raw/web_app_manifest.json` on ChromeOS, not from the Intent used to start the app? (I don't have a Chrome OS device to test).

@nohe427 @joycetoh8 FYI

**Comment by jeevan-shikaram** [comment]
@andreban: You're correct, the `start_url` is being extracted from the `manifest.json` in `res/raw`.

@kinyat: On first run, how are you launching the app? Via an Android intent (clicking open on the Play Store) or clicking the icon in the shelf? If launching via an Android intent, make sure that the intent's target URL includes the start parameters. If launching via the shelf, I'll need to look into why it's not showing the query params.

@dominickng FYI

**Comment by kinyat** [comment]
> @kinyat: On first run, how are you launching the app? Via an Android intent (clicking open on the Play Store) or clicking the icon in the shelf? If launching via an Android intent, make sure that the intent's target URL includes the start parameters. If launching via the shelf, I'll need to look into why it's not showing the query params.

On the first run, it was via an Android intent (i.e. via the Play Store dialog after installation) which didn't have query parameter set at all no matter if both `startUrl` and `start_url` had query parameters. 
Clicking the icon in the shelf would have the query parameter attached if `start_url` in the webmanifest file had the query parameter set.

**Comment by jeevan-shikaram** [comment]
@andreban it looks like we need to make sure that the Android intent will have the start_url query params as well.

**Comment by andreban** [comment]
@jeevan-shikaram do we mean the Intent used by LauncherActivity to open Chrome when the app is started from the home screen? That one should already have the `start_url` parameter, as defined in `twa-manifest.json`.

**Comment by PEConn** [comment]
Do remember the start_url parameter won't be present when the TWA is launched as the result of the user following a link that launches the TWA.

**Comment by kinyat** [comment]
May I ask if there are any follow-ups with this issue?

We have come into issues where we want to promote the Google Payment option when the user is under TWA so this becomes a blocker to us.

Thanks again for the support!

**Comment by PEConn** [comment]
I think the ChromeOS team are planning on working on this, but haven't got around to it yet.

For the use case of TWAs and Play Billing, there's a better solution though:

```
const PAYMENT_METHOD = "https://play.google.com/billing";
await window.getDigitalGoodsService(PAYMENT_METHOD);
```

If your code is not running in a TWA, this will return `null`, so my advice is to just check to see if you can get ahold of this object.

(Unfortunately this won't work on versions of Chrome where the Digital Goods API hasn't been released yet.)

**Comment by jeevan-shikaram** [comment]
@kinyat just wanted to let you know that we are working on this and will get a fix published as soon as we can.

**Comment by jeevan-shikaram** [comment]
Hi @kinyat, we've merged in a potential fix pull request #440. Please test it out and let us know if you are still experiencing problems.

**Comment by kinyat** [comment]
@jeevan-shikaram Thanks for the fix. It seems working fine for me now. I will close this issue unless it is coming back :)
